### PR TITLE
Clean shutdown of some auth integration tests

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref https://github.com/kubernetes/kubernetes/issues/108483

It's fixing tests only from auth/auth_test.go. But the change is large enough I decided to split it up to a separate PR.

```release-note
NONE
```

/kind cleanup
/priority important-longterm
/sig testing

### CODE REVIEW

File: test/integration/auth/auth_test.go [hunk 1]
```diff
@@ -486,31 +514,26 @@ func TestAuthModeAlwaysAllow(t *testing.T) {
...
-		if r.verb == "PATCH" {
-			req.Header.Set("Content-Type", "application/merge-patch+json")
+			t.Fatal(err)
 		}
 		func() {
-			resp, err := transport.RoundTrip(req)
-			if err != nil {
+			resp := req.Do(context.Background())
+			// We're consciously ignoring the error here, as error is set for
+			// non-success status codes, which we're exercising in different tests.
```
Comment (aojea) [inline test/integration/auth/auth_test.go:525]: > // StatusCode returns the HTTP status code of the request. (Only valid if no
> // error was returned.)
> func (r Result) StatusCode(statusCode *int) Result {

Comment (aojea) [reply test/integration/auth/auth_test.go:525]: I guess it will fail on the map lookup later :thinking:

File: test/integration/auth/auth_test.go [hunk 2]
```diff
@@ -336,18 +337,35 @@ func addTimeoutFlag(URLString string) string {
...
+
+func (r *testRequest) restRequest(c clientset.Interface) (*rest.Request, error) {
+	bodyBytes := bytes.NewReader([]byte(r.body))
+
+	u, err := url.Parse(r.URL)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse URL: %v", err)
+	}
+
+	req := c.CoreV1().RESTClient().Verb(r.verb).Body(bodyBytes).AbsPath(u.Path)
```
Comment (aojea) [inline test/integration/auth/auth_test.go:355]: is better to use client-go instead the vanilla http client? 
it adds a things like retries ...

**Comment by aojea** [comment]
just one question, https://github.com/kubernetes/kubernetes/pull/110505#discussion_r894629518

is this change to client-go required for the next PR?

File: test/integration/auth/auth_test.go (hunk 2 above)
Comment (wojtek-t) [reply test/integration/auth/auth_test.go:355]: Yeah - I realized that I overcomplicated after I sent this out.

Let me try redoing that to make the diff smaller.

Comment (wojtek-t) [reply test/integration/auth/auth_test.go]: Fixed

**Comment by wojtek-t** [comment]
/hold cancel

@aojea - PTAL

**Comment by aojea** [comment]
/lgtm

great, thanks

**Pull request merged**